handshake_client.go

```diff
@@ -86,7 +86,12 @@ NextCipherSuite:
 			// Don't advertise TLS 1.2-only cipher suites unless
 			// we're attempting TLS 1.2.
 			if hello.vers < VersionTLS12 && suite.flags&suiteTLS12 != 0 {
-				continue
+				continue NextCipherSuite
+			}
+			// Don't advertise TLS 1.3-only cipher suites unless
+			// we're attempting TLS 1.3.
+			if hello.vers < VersionTLS13 && suite.flags&suiteTLS13 != 0 {
+				continue NextCipherSuite
```

Comment:
Shouldn't both of these just be a normal `continue`? Otherwise it will not try to match other potential cipher suites from `cipherSuites` against the proposed candidate from the configuration (`possibleCipherSuites`) once it notices an incompatible cipher suite.

Reply:
@Lekensteyn I'm pretty sure the logic is correct. It's not hugely important as it's really nothing more than an optimisation.

The outer loop is the configured cipher suites and the inner loop is the crypto/tls supported cipher suites. That's the desired behaviour because cipher suites have unique ids so no more than one suite (from `cipherSuites`) can ever match. (Unless I'm fatally misunderstanding something here).